python/lsst/multiprofit/modeller.py

```diff
@@ -0,0 +1,840 @@
+        title="Jacobian arrays (views) for each observation",
+    )
+    outputs_prior: tuple[g2.ImageD, ...] = pydantic.Field(
+        title="Jacobian arrays (views) for each free parameter's prior",
+    )
+    residual: np.ndarray = pydantic.Field(title="The full residual (chi) array")
+    residuals: list[g2.ImageD] = pydantic.Field(
+        default_factory=list,
+        title="Residual (chi) arrays (views) for each observation",
+    )
+    residuals_prior: g2.ImageD = pydantic.Field(
+        title="Shared residual array for all Prior instances",
+    )
+
+    @classmethod
+    def get_sizes(
+        cls,
+        model: Model,
+    ):
+        """Initialize Jacobian and residual arrays for a model.
+
+        Parameters
+        ----------
+        model : `gauss2d.fit.Model`
+            The model to initialize arrays for.
+        """
+        priors = model.priors
+        n_prior_residuals = sum(len(p) for p in priors)
+        params_free = tuple(get_params_uniq(model, fixed=False))
+        n_params_free = len(params_free)
+        # There's one extra validation array
+        n_params_jac = n_params_free + 1
+        if not (n_params_jac > 1):
+            raise ValueError("Can't fit model with no free parameters")
+
+        n_obs = len(model.data)
+        shapes = np.zeros((n_obs, 2), dtype=int)
+        ranges_params = [None] * n_obs
+
+        for idx_obs in range(n_obs):
+            observation = model.data[idx_obs]
+            shapes[idx_obs, :] = (observation.image.n_rows, observation.image.n_cols)
+            params = tuple(get_params_uniq(model, fixed=False, channel=observation.channel))
+            n_params_obs = len(params)
+            ranges_params_obs = [0] * (n_params_obs + 1)
+            for idx_param in range(n_params_obs):
+                ranges_params_obs[idx_param + 1] = params_free.index(params[idx_param]) + 1
+            ranges_params[idx_obs] = ranges_params_obs
+
+        n_free_first = len(ranges_params[0])
+        assert all([len(rp) == n_free_first for rp in ranges_params[1:]])
+
+        return n_obs, n_params_jac, n_prior_residuals, shapes
+
+    @classmethod
+    def from_model(
+        cls,
+        model: Model,
+    ):
+        """Initialize Jacobian and residual arrays for a model.
+
+        Parameters
+        ----------
+        model : `gauss2d.fit.Model`
+            The model to initialize arrays for.
+        """
+        n_obs, n_params_jac, n_prior_residuals, shapes = cls.get_sizes(model)
+        n_data_obs = np.cumsum(np.prod(shapes, axis=1))
+        n_data = n_data_obs[-1]
+        size_data = n_data + n_prior_residuals
+        shape_jacobian = (size_data, n_params_jac)
+        jacobian = np.zeros(shape_jacobian)
+        jacobians = [None] * n_obs
+        outputs_prior = [None] * n_params_jac
+        for idx in range(n_params_jac):
+            outputs_prior[idx] = g2.ImageD(jacobian[n_data:, idx].view().reshape((1, n_prior_residuals)))
```

Comment:
Well, most types of slicing should, but apparently some don't. However, I think you're correct anyway since it's the slicing that determines whether it's a view or not and calling view on it won't accomplish anything.